examples/view-pure-read/contracts/ReadViewOrPure.sol

```diff
@@ -0,0 +1,180 @@
+                payable(msg.sender)
+            );
+    }
+
+    /**
+     * @notice Estimates the messaging fee required to perform the read operation.
+     *
+     * @param a The first number to sum.
+     * @param b The second number to sum.
+     * @param _extraOptions Additional messaging options.
+     *
+     * @return fee The estimated messaging fee.
+     */
+    function quoteReadFee(
+        uint256 a,
+        uint256 b,
+        bytes calldata _extraOptions
+    ) external view returns (MessagingFee memory fee) {
+        return _quote(READ_CHANNEL, _getCmd(a, b), _combineOptions(READ_CHANNEL, READ_TYPE, _extraOptions), false);
+    }
+
+    /**
+     * @notice Constructs the read command to call the `add` function.
+     *
+     * @param a The first number.
+     * @param b The second number.
+     *
+     * @return The encoded command.
+     */
+    function _getCmd(uint256 a, uint256 b) internal view returns (bytes memory) {
+        // Encode callData to call add(a, b) function using the function selector
+        bytes memory callData = abi.encodeWithSelector(IExampleContract.add.selector, a, b);
+
+        // Create an array of EVMCallRequestV1 with a single read request
+        EVMCallRequestV1[] memory readRequests = new EVMCallRequestV1[](1);
+        readRequests[0] = EVMCallRequestV1({
+            appRequestLabel: 1, // Application-specific label for tracking
+            targetEid: targetEid, // Endpoint ID of the target chain
+            isBlockNum: false, // Use timestamp instead of block number
+            blockNumOrTimestamp: uint64(block.timestamp), // Timestamp to read the state at
+            confirmations: 15, // Number of confirmations to wait for finality
+            to: targetContractAddress, // Address of the contract to call
+            callData: callData // Encoded function call data
+        });
+
+        // No compute logic needed; encode the command with appLabel 0
+        return ReadCodecV1.encode(0, readRequests);
+    }
+
+    /**
+     * @notice Handles the received sum from the target chain.
+     *
+     * @dev This function is called internally by the LayerZero protocol.
+     *
+     * @param _message The encoded sum received.
+     */
+    function _lzReceive(
+        Origin calldata /*_origin*/,
+        bytes32 /*_guid*/,
+        bytes calldata _message,
+        address /*_executor*/,
+        bytes calldata /*_extraData*/
+    ) internal override {
+        // Decode the sum received
+        require(_message.length == 32, "Invalid message length");
+        uint256 sum = abi.decode(_message, (uint256));
+        emit SumReceived(sum);
+    }
+
+    /**
+     * @notice Combines the options for messaging.
+     *
+     * @param _channelId The channel ID.
+     * @param _msgType The message type.
+     * @param _extraOptions Additional options.
+     *
+     * @return options The combined options.
+     */
+    function _combineOptions(
```

Comment:
Why is this needed v.s. just inlining calls to OAppOptionsType3.combineOptions(...)?